README.md

```diff
@@ -1,47 +1,185 @@
 # react-e [![CircleCI](https://circleci.com/gh/Originate/react-e.svg?style=svg)](https://circleci.com/gh/Originate/react-e)
 
-React helper to generate elements
+JSX replacement that doesn't need to be compiled.
 
-## Installing
-`npm install --save react-e`
+## Basic Usage
 
-## Example Usage
+```javascript
+e = require('react-e')
 
-```coffeescript
-e = require 'react-e'
-
-e '.container',
-  e 'button.primary', 'Click me'
-  e 'span.text', className: active: true, 'Some text'
+e(Modal,
+  e('.container',
```

Comment:
I'm still not a fan of the redundant markup. For example, to create a button, I need `e` and `Button`. Why can't `Button` be enough here? Even JSX does that with one piece of boilerplate. 

Here is how an optimized syntax would look like in Kevin's dream world inside his head:

```js
{$, Button, Modal, span} = require('react-e').domElements
// or $ = require('react-e') for the old way

Modal(
  $('.container',
    span('.text', {className: {active: this.state.active}}, 'Some text'),
    span({className: ['text', {active: this.state.active}]}, 'Some text'),  // alternative syntax
    Button({className: 'primary'}, 'Close')
  )
)
```

Just an idea  :)